README.md
<p align="center">
  <img width="200px" src="https://github.com/yeukfei02/lunchPickerReactNative/blob/master/readme-icon.png"><br/>
  <h2 align="center">lunchPickerReactNative</h2>
</p>

<p align="center">
  <a href="https://codecov.io/gh/yeukfei02/lunchPickerReactNative"><img src="https://codecov.io/gh/yeukfei02/lunchPickerReactNative/branch/master/graph/badge.svg" alt=""></a>
  <a href="https://discord.gg/ZSwskB2"><img src="https://img.shields.io/discord/709280499503202364" alt=""></a>
</p>

If you’re a working software engineer, you’ve probably encountered with one of the toughest questions, where should I have lunch?

Lunch picker is the tool you’ll turn to to answer this question.

<p align="center">
  <a href="https://apps.apple.com/us/app/lunchpickerlite/id1509385881"><img src="https://github.com/yeukfei02/lunchPickerReactNative/blob/master/app-store-badge.png" width="30%" height="30%" alt=""></a>
  <a href="https://play.google.com/store/apps/details?id=com.donaldwu.lunchpickerlite"><img src="https://github.com/yeukfei02/lunchPickerReactNative/blob/master/google-play-badge.png" width="30%" height="30%" alt=""></a>
</p>

## Requirement

- install expo-cli
- install eas-cli
- install yarn
- install node (v14+)

## Testing and run

```zsh
$ yarn

// expo start
$ yarn run start

// expo start android
$ yarn run android

// expo start ios
$ yarn run ios

// run test case
$ yarn run test

// lint code
$ yarn run lint

// format code
$ yarn run format
```

```zsh
// publish project to expo
$ expo publish

// make ios ipa
$ expo build:ios

// make android apk / app bundle
$ expo build:android

// get ios certificate
$ expo fetch:ios:certs

// get android keystore
$ expo fetch:android:keystore

// get current build status
$ expo build:status

// generate android + ios folder
$ expo eject

// eas build ios/android
$ eas build
```

## Contributing

Please refer to [CONTRIBUTING.md](https://github.com/yeukfei02/lunchPickerReactNative/blob/master/CONTRIBUTING.md)


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

## [1.9.0](https://github.com/yeukfei02/lunchPickerReactNative/compare/v1.8.2...v1.9.0) (2022-07-13)


### Features

* 🎸 add input field for searchText ([ad3263c](https://github.com/yeukfei02/lunchPickerReactNative/commit/ad3263cb6f19ae54d6667640e94dd50a02a06cdc))
* 🎸 add lunchpicker web in contact us ([ee57e8f](https://github.com/yeukfei02/lunchPickerReactNative/commit/ee57e8fe1a35239eba50eaa6f11f6113872b333f))

### [1.8.2](https://github.com/yeukfei02/lunchPickerReactNative/compare/v1.8.1...v1.8.2) (2022-06-17)


### Bug Fixes

* 🐛 update eas.json ([9c7f155](https://github.com/yeukfei02/lunchPickerReactNative/commit/9c7f1559eac038f836e0dc110257f165d3a1d161))
* 🐛 update env var ([3ebbdef](https://github.com/yeukfei02/lunchPickerReactNative/commit/3ebbdef7fe2995273e5196310e5e5215564254f4))
* 🐛 update version ([74aa27d](https://github.com/yeukfei02/lunchPickerReactNative/commit/74aa27d71e310ac83a232ccfd30e2ea59fb100b1))

### [1.8.1](https://github.com/yeukfei02/lunchPickerReactNative/compare/v1.8.0...v1.8.1) (2022-06-17)


### Bug Fixes

* 🐛 update app.json ([b5c3bb4](https://github.com/yeukfei02/lunchPickerReactNative/commit/b5c3bb47a961221432df90c24debbdb622eb0218))
* 🐛 update eas.json env ([4525101](https://github.com/yeukfei02/lunchPickerReactNative/commit/4525101003bf854b15ad6b607557679dba056bab))
* 🐛 update version ([13bfe89](https://github.com/yeukfei02/lunchPickerReactNative/commit/13bfe89b94a82bdc69228b903341e798efad8bb7))

## [1.8.0](https://github.com/yeukfei02/lunchPickerReactNative/compare/v1.7.0...v1.8.0) (2022-06-17)


### Features

* 🎸 add eas.json ([511c90a](https://github.com/yeukfei02/lunchPickerReactNative/commit/511c90a3d9949f1e6918fd2e54f637c39b0ddcff))


### Bug Fixes

* 🐛 remove not used engine ([4604010](https://github.com/yeukfei02/lunchPickerReactNative/commit/4604010cea4f2a8344dbdd96e7f41db015e61be7))
* 🐛 update version ([b709838](https://github.com/yeukfei02/lunchPickerReactNative/commit/b7098382c1b41cfeaca9d7e243146eb199df02e7))

## [1.7.0](https://github.com/yeukfei02/lunchPickerReactNative/compare/v1.6.7...v1.7.0) (2022-06-17)


### Features

* 🎸 add .env.sample ([557c3d6](https://github.com/yeukfei02/lunchPickerReactNative/commit/557c3d6d6801fcf965db73d0dad03a56617e0dd3))
* 🎸 add format command and update husky ([99203f3](https://github.com/yeukfei02/lunchPickerReactNative/commit/99203f3e51f54d9b03ab1075997b19d7cb8df0ef))
* 🎸 add semgrep.yml ([cdcdc63](https://github.com/yeukfei02/lunchPickerReactNative/commit/cdcdc635288c0e0cdb08ae4f9e632ca0a5e04c57))


### Bug Fixes

* 🐛 add engine ([f6d0a1f](https://github.com/yeukfei02/lunchPickerReactNative/commit/f6d0a1f4f2aed6d4a8b6a6397c4b13025d29a3a1))
* 🐛 remove not used travis ([0a40e5a](https://github.com/yeukfei02/lunchPickerReactNative/commit/0a40e5ad0b81b0981d1e00fb8f7cf919d341f9bb))
* 🐛 update expo version, other lib, logic ([51835b0](https://github.com/yeukfei02/lunchPickerReactNative/commit/51835b0de8149391d7f048bf6cc72fe5b490a7be))

### [1.6.7](https://github.com/yeukfei02/lunchPickerReactNative/compare/v1.6.6...v1.6.7) (2021-11-16)


### Bug Fixes

* 🐛 update to latest sentry ([26e85e0](https://github.com/yeukfei02/lunchPickerReactNative/commit/26e85e0951efb1e152b04d3b18449d7ee333e363))

### [1.6.6](https://github.com/yeukfei02/lunchPickerReactNative/compare/v1.6.5...v1.6.6) (2021-11-12)


### Bug Fixes

* 🐛 remove log func in common ([5f1dcba](https://github.com/yeukfei02/lunchPickerReactNative/commit/5f1dcba21b7659d948fc48c177c23f720267a547))
* 🐛 rename to helpers ([e5af121](https://github.com/yeukfei02/lunchPickerReactNative/commit/e5af1213f07171ffb893539f966f8cb14b5baafd))
* 🐛 update app.json ([0511333](https://github.com/yeukfei02/lunchPickerReactNative/commit/05113334492b31de2d132e10d3bfe3fb5d63a6a0))
* 🐛 update react-native-screens lib ([bb0313e](https://github.com/yeukfei02/lunchPickerReactNative/commit/bb0313e14be2c5ed6ba559f22b14ae3bd9c7ddca))

### [1.6.5](https://github.com/yeukfei02/lunchPickerReactNative/compare/v1.6.4...v1.6.5) (2021-03-15)

### [1.6.4](https://github.com/yeukfei02/lunchPickerReactNative/compare/v1.6.3...v1.6.4) (2021-03-15)


### Bug Fixes

* 🐛 remove sentry-expo hook ([8d7b187](https://github.com/yeukfei02/lunchPickerReactNative/commit/8d7b1874ba05e914d95ea3186249084180cc43ee))
* 🐛 update version in app.json ([b040de2](https://github.com/yeukfei02/lunchPickerReactNative/commit/b040de253d7a1d7c312239b289ae95f680967533))

### [1.6.3](https://github.com/yeukfei02/lunchPickerReactNative/compare/v1.6.2...v1.6.3) (2021-01-28)


### Bug Fixes

* 🐛 fix package.json ([96edd78](https://github.com/yeukfei02/lunchPickerReactNative/commit/96edd7812ebc748c59c4ac41f0cc78cc40224f69))

### [1.6.2](https://github.com/yeukfei02/lunchPickerReactNative/compare/v1.6.1...v1.6.2) (2020-12-28)


### Bug Fixes

* 🐛 hide fab button in home component ([799cacd](https://github.com/yeukfei02/lunchPickerReactNative/commit/799cacdde89ab31e3dd9c36a1b8439db7ad08908))

### [1.6.1](https://github.com/yeukfei02/lunchPickerReactNative/compare/v1.5.10...v1.6.1) (2020-12-18)

### [1.5.10](https://github.com/yeukfei02/lunchPickerReactNative/compare/v1.5.9...v1.5.10) (2020-12-18)


### Bug Fixes

* 🐛 add KeyboardAwareScrollView to avoid keyboard cover ([667dff3](https://github.com/yeukfei02/lunchPickerReactNative/commit/667dff32498114f41dc772c109b29777bde41d01))

### [1.5.9](https://github.com/yeukfei02/lunchPickerReactNative/compare/v1.5.8...v1.5.9) (2020-12-16)


### Bug Fixes

* 🐛 fix keyboardDismiss after scrollview scroll ([47d6e41](https://github.com/yeukfei02/lunchPickerReactNative/commit/47d6e41806673e502049040e1206f14bbcb91e49))

### [1.5.8](https://github.com/yeukfei02/lunchPickerReactNative/compare/v1.5.7...v1.5.8) (2020-12-15)


### Bug Fixes


src/components/contactUs/ContactUs.js
import React, { useState, useEffect } from 'react';
import { StyleSheet, ScrollView, View, Text, Linking } from 'react-native';
import { RadioButton, Checkbox, Button, TextInput, Card, Snackbar } from 'react-native-paper';
import { Dropdown } from 'react-native-material-dropdown';
import { AntDesign, MaterialIcons } from '@expo/vector-icons';
import { SliderBox } from "react-native-image-slider-box";
import { Table, Row, Rows } from 'react-native-table-component';
import { LiteCreditCardInput } from "react-native-credit-card-input";
import Stripe from 'react-native-stripe-api';
import { useTranslation } from 'react-i18next';
import _ from 'lodash';
import axios from 'axios';
import { getRootUrl, log, getStripeApiKey } from '../../common/Common';

import Divder from '../divider/Divider';

const ROOT_URL = getRootUrl();

const style = StyleSheet.create({
  scrollViewContainer: {
    flex: 1,
    backgroundColor: '#FAFAD2',
  },
  container: {
    flex: 1,
    marginTop: 100,
    padding: 20,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'white',
    marginHorizontal: 30
  },
  titleStyle: {
    fontSize: 18,
    fontWeight: 'bold'
  },
  iconContainer: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  donateCardViewContainer: {
    flex: 1,
    padding: 20,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'white',
    marginVertical: 35,
    marginHorizontal: 30
  },
  rowContainer: {
    flexDirection: 'row',
    alignSelf: 'flex-start'
  },
  cardViewContainer: {
    flex: 1,
    padding: 20,
    backgroundColor: 'white',
    marginVertical: 15,
    marginHorizontal: 30
  },
  restaurantDetailsTitleText: {
    fontSize: 16,
    fontWeight: 'bold'
  },
  restaurantDetailsValueText: {
    fontSize: 16,
    fontWeight: 'normal'
  },
  restaurantDetailsUrlValueText: {
    fontSize: 16,
    fontWeight: 'normal',
    color: '#ed1f30',
  },
  restaurantDetailsLocationValueText: {
    fontSize: 16,
    fontWeight: 'normal',
    color: '#ed1f30',
    textDecorationLine: 'underline'
  },
  tableHead: {
    height: 40,
    backgroundColor: '#ed1f30',
  },
  tableHeadText: {
    margin: 6,
    color: 'white'
  },
  tableRowText: {
    margin: 6,
  },
  colorPrimary: {
    color: '#ed1f30'
  }
});

function RestaurantDetails({ navigation, id }) {
  const { t } = useTranslation();

  const [restaurantDetails, setRestaurantDetails] = useState({});
  const [photosList, setPhotosList] = useState([]);
  const [name, setName] = useState('');
  const [locationStr, setLocationStr] = useState('');

  useEffect(() => {
    if (!_.isEmpty(id)) {
      getRestaurantsDetailsById(id);
    }
  }, [id]);

  const getRestaurantsDetailsById = (id) => {
    axios.get(
      `${ROOT_URL}/restaurant/get-restaurant-details/${id}`,
      {
        headers: {
          'Content-Type': 'application/json'
        }
      }
    )
      .then((response) => {
        if (!_.isEmpty(response)) {
          log("response = ", response);
          setRestaurantDetails(response.data.restaurantDetails);

          const name = response.data.restaurantDetails.name;
          setName(name);

          const photos = response.data.restaurantDetails.photos;
          setPhotosList(photos);

          const location = response.data.restaurantDetails.location;
          let locationStr = '';
          if (!_.isEmpty(location)) {
            if (!_.isEmpty(location.display_address)) {
              locationStr = location.display_address.join(', ');
            }
          }
          setLocationStr(locationStr);
        }
      })
      .catch((error) => {
        if (!_.isEmpty(error)) {
          log("error = ", error);
        }
      });
  }

  const handleBackToHome = () => {
    navigation.navigate(t('home'));
  }

  const handleOpenUrl = () => {
    Linking.openURL(`${restaurantDetails.url}`);
  }

  const handleLocationClick = () => {
    Linking.openURL(`https://www.google.com/maps/search/?api=1&query=${locationStr}`);
  }

  const renderOpeningTimeTable = () => {
    let table = null;

    let formattedDataList = [];
    let hoursType = '';
    let isOpenNow = '';

    if (!_.isEmpty(restaurantDetails)) {
      const hours = restaurantDetails.hours;
      if (!_.isEmpty(hours)) {
        hours.forEach((item, i) => {
          const open = item.open;
          if (!_.isEmpty(open)) {
            open.forEach((item, i) => {
              let data = [];

              switch (item.day) {
                case 0:
                  item.day = "Mon";
                  break;
                case 1:
                  item.day = "Tue";
                  break;
                case 2:
                  item.day = "Wed";
                  break;
                case 3:
                  item.day = "Thu";
                  break;
                case 4:
                  item.day = "Fri";
                  break;
                case 5:
                  item.day = "Sat";
                  break;
                case 6:
                  item.day = "Sun";
                  break;
                default:

              }
              if (!item.start.includes(':')) {
                item.start = `${item.start.substring(0, 2)}:${item.start.substring(2)}`;
              }
              if (!item.end.includes(':')) {
                item.end = `${item.end.substring(0, 2)}:${item.end.substring(2)}`;
              }
              item.is_overnight = item.is_overnight ? 'yes' : 'no';

              data.push(item.day);
              data.push(item.start);
              data.push(item.end);
              data.push(item.is_overnight);
              formattedDataList.push(data);
            });
          }

          hoursType = item.hours_type;
          isOpenNow = item.is_open_now;
        });
      }
    }

    const tableHead = [
      'Days',
      'Start',
      'End',
      'Is overnight'
    ];
    const tableData = formattedDataList;
    if (!_.isEmpty(tableHead) && !_.isEmpty(tableData)) {
      table = (
        <Card style={style.cardViewContainer}>
          <Table borderStyle={{ borderWidth: 1.5, borderColor: 'black' }}>
            <Row data={tableHead} style={style.tableHead} textStyle={style.tableHeadText} />
            <Rows data={tableData} textStyle={style.tableRowText} />
          </Table>

          <Divder margin={10} />
          <Text style={style.titleStyle}>{t('hoursType')} <Text style={{ fontWeight: 'normal', color: style.colorPrimary.color }}>{hoursType.toLowerCase()}</Text></Text>
          <Divder margin={8} />
          <View style={style.rowContainer}>
            <Checkbox
              status={isOpenNow ? 'checked' : 'unchecked'}
              disabled={true}
            />
            <Text style={{ fontSize: 16, marginTop: 8, marginLeft: 5 }}>{t('isOpenNow')}</Text>
          </View>
          <Divder margin={10} />
          <Button style={{ alignSelf: 'stretch' }} mode="outlined" color={style.colorPrimary.color} onPress={handleBackToHome}>
            {t('backToHome')}
          </Button>
        </Card>
      );
    }

    return table;
  }

  return (
    <View>
      <View style={{ marginTop: 100 }}>
        <SliderBox
          images={photosList}
          sliderBoxHeight={250}
          dotColor={style.colorPrimary.color}
          inactiveDotColor="lightgray" />
      </View>
      <Divder margin={8} />
      <Card style={style.cardViewContainer}>
        <Text style={style.titleStyle}>{t('restaurantDetails')}</Text>
        <Divder margin={10} />
        <Text style={style.restaurantDetailsTitleText}>{t('name')} <Text style={style.restaurantDetailsValueText}>{name}</Text></Text>
        <Divder margin={10} />
        <Text style={style.restaurantDetailsTitleText}>{t('phone')} <Text style={style.restaurantDetailsValueText}>{restaurantDetails.phone}</Text></Text>
        <Divder margin={10} />
        <Text style={style.restaurantDetailsTitleText}>{t('url')} <Text style={style.restaurantDetailsUrlValueText} onPress={handleOpenUrl}>Open Url</Text></Text>
        <Divder margin={10} />
        <Text style={style.restaurantDetailsTitleText}>{t('location')} <Text style={style.restaurantDetailsLocationValueText} onPress={handleLocationClick}>{locationStr}</Text></Text>
      </Card>
      {renderOpeningTimeTable()}
    </View >
  );
}

function ContactUs({ navigation, route }) {
  const { t } = useTranslation();

  const [radioButtonValue, setRadioButtonValue] = useState('stripe');

  const [currencyList, setCurrencyList] = useState([]);
  const [amount, setAmount] = useState('0');
  const [currency, setCurrency] = useState('');
  const [token, setToken] = useState('');
  const [card, setCard] = useState({});

  const [cardValid, setCardValid] = useState(false);
  const [cardInfoData, setCardInfoData] = useState(null);

  const [payNowButtonClicked, setPayNowButtonClicked] = useState(false);
  const [snackBarStatus, setSnackBarStatus] = useState(false);
  const [snackBarMessage, setSnackBarMessage] = useState('');

  useEffect(() => {
    getCurrencyList();
  }, []);

  useEffect(() => {
    if (!_.isEmpty(amount) && !_.isEmpty(currency) && !_.isEmpty(token) && !_.isEmpty(card)) {
      creditCardPayment(amount, currency, token, card);
      setToken('');
      setCard({});
    }
  }, [amount, currency, token, card]);

  const getCurrencyList = () => {
    const currencyList = [
      { value: 'Hong Kong Dollar (HKD)' },
      { value: 'Singapore Dollar (SGD)' },
      { value: 'British Dollar Pound (GBP)' },
      { value: 'Chinese Renminbi Yuan (CNY)' },
      { value: 'US Dollar (USD)' },
    ];
    setCurrencyList(currencyList);
  }

  const handleRadioButton = (radioButtonValue) => {
    setRadioButtonValue(radioButtonValue);
  }

  const handleGithubClick = () => {
    Linking.openURL(`https://github.com/yeukfei02`);
  }

  const handleEmailClick = () => {
    Linking.openURL(`mailto:yeukfei02@gmail.com`);
  }

  const handleDonorboxClick = () => {
    Linking.openURL('https://donorbox.org/donate-for-lunch-picker-better-features-and-development');
  }

  const handleBuyMeACoffeeClick = () => {
    Linking.openURL('https://www.buymeacoffee.com/yeukfei02');
  }

  const handlePayNow = async () => {
    setPayNowButtonClicked(true);

    try {
      if (!_.isEmpty(amount) && !_.isEmpty(currency) && cardValid) {
        const stripeApiKey = getStripeApiKey();
        const apiKey = stripeApiKey;
        const client = new Stripe(apiKey);

        const token = await client.createToken({
          number: cardInfoData.values.number,
          exp_month: cardInfoData.values.expiry.substring(0, 2),
          exp_year: cardInfoData.values.expiry.substring(3),
          cvc: cardInfoData.values.cvc,
        });
        if (!_.isEmpty(token)) {
          setToken(token.id);
          setCard(token.card);

          setPayNowButtonClicked(false);
          setSnackBarStatus(true);
          setSnackBarMessage(`Payment success`);
        }
      } else {
        setPayNowButtonClicked(false);
        setSnackBarStatus(true);
        setSnackBarMessage(`Please check amount/currency/card input`);
      }
    } catch (e) {
      setPayNowButtonClicked(false);
      setSnackBarStatus(true);
      setSnackBarMessage(`Payment error = ${e.message}`);
    }
  }

  const renderSelectDropdown = () => {
    let selectDropdown = null;

    const data = getDropdownData();
    if (!_.isEmpty(data)) {
      selectDropdown = (
        <Dropdown
          label={t('selectCurrency')}
          data={data}
          onChangeText={handleCurrencyChange}
        />
      );
    }

    return selectDropdown;
  }

  const renderResultDiv = () => {
    let resultDiv = null;

    if (_.isEqual(radioButtonValue, 'donorbox')) {
      resultDiv = (
        <Button style={{ alignSelf: 'stretch' }} mode="contained" color={style.colorPrimary.color} onPress={handleDonorboxClick}>
          Donorbox
        </Button>
      );
    } else if (_.isEqual(radioButtonValue, 'buyMeACoffee')) {
      resultDiv = (
        <Button style={{ alignSelf: 'stretch' }} mode="contained" color={style.colorPrimary.color} onPress={handleBuyMeACoffeeClick}>
          Buy Me A Coffee
        </Button>
      );
    } else if (_.isEqual(radioButtonValue, 'stripe')) {
      resultDiv = (
        <View style={{ alignSelf: 'stretch' }}>
          <TextInput
            mode="outlined"
            label={t('amount')}
            value={amount}
            placeholder={t('enterAmount')}
            onChangeText={(number) => handleAmountChange(number)}
          />
          <Divder margin={5} />
          {renderSelectDropdown()}
          <Divder margin={5} />
          <LiteCreditCardInput onChange={handleCreditCardInputChange} />
          <Divder margin={5} />
          {renderPaynowButton()}
        </View>
      );
    }

    return resultDiv;
  }

  const getDropdownData = () => {
    return currencyList;
  }

  const renderPaynowButton = () => {
    let paynowButton = (
      <Button style={{ alignSelf: 'stretch' }} mode="contained" color={style.colorPrimary.color} onPress={handlePayNow}>
        {t('payNow')}
      </Button>
    );

    if (payNowButtonClicked) {
      paynowButton = (
        <Button style={{ alignSelf: 'stretch' }} mode="contained" color={style.colorPrimary.color} disabled={true} onPress={handlePayNow}>
          {t('loading')}
        </Button>
      );
    }

    return paynowButton;
  }

  const handleAmountChange = (number) => {
    if (!isNaN(number))
      setAmount(number);
  }

  const handleCurrencyChange = (selectedCurrency) => {
    if (!_.isEmpty(selectedCurrency)) {
      switch (selectedCurrency) {
        case 'Hong Kong Dollar (HKD)':
          setAmount('3');
          setCurrency('hkd');
          break;
        case 'Singapore Dollar (SGD)':
          setAmount('1');
          setCurrency('sgd');
          break;
        case 'British Dollar Pound (GBP)':
          setAmount('1');
          setCurrency('gbp');
          break;
        case 'Chinese Renminbi Yuan (CNY)':
          setAmount('3');
          setCurrency('cny');
          break;
        case 'US Dollar (USD)':
          setAmount('1');
          setCurrency('usd');
          break;
        default:

      }
    }
  }

  const handleCreditCardInputChange = (inputData) => {
    if (!_.isEmpty(inputData)) {
      if (inputData.valid) {
        setCardInfoData(inputData);
        setCardValid(true);
      } else {
        setCardInfoData(null);
        setCardValid(false);
      }
    }
  }

  const creditCardPayment = (amount, currency, token, card) => {
    axios.post(
      `${ROOT_URL}/stripe/credit-card-payment`,
      {
        amount: Math.round(amount * 100),
        currency: currency,
        token: token,
        card: card
      },
      {
        headers: {
          'Content-Type': 'application/json'
        }
      }
    )
      .then((response) => {
        if (!_.isEmpty(response)) {
          log("response = ", response);
        }
      })
      .catch((error) => {
        if (!_.isEmpty(error)) {
          log("error = ", error);
        }
      });
  }

  const renderDiv = () => {
    let result = (
      <View>
        <Card style={style.container}>
          <Text style={style.titleStyle}>{t('contactTitle')}</Text>
          <Divder margin={5} />
          <View style={style.iconContainer}>
            <AntDesign style={{ marginRight: 15 }} name="github" size={40} color="black" onPress={handleGithubClick} />
            <MaterialIcons name="email" size={40} color="black" onPress={handleEmailClick} />
          </View>
        </Card>

        <Card style={style.donateCardViewContainer}>
          <Text style={style.titleStyle}>{t('donateTitle')}</Text>

          <Divder margin={5} />

          <View style={style.rowContainer}>
            <RadioButton
              color={style.colorPrimary.color}
              value="places"
              status={radioButtonValue === 'donorbox' ? 'checked' : 'unchecked'}
              onPress={() => handleRadioButton('donorbox')}
            />
            <Text style={{ marginTop: 8, marginLeft: 8 }}>Donorbox</Text>
          </View>

          <Divder margin={5} />

          <View style={style.rowContainer}>
            <RadioButton
              color={style.colorPrimary.color}
              value="places"
              status={radioButtonValue === 'buyMeACoffee' ? 'checked' : 'unchecked'}
              onPress={() => handleRadioButton('buyMeACoffee')}
            />
            <Text style={{ marginTop: 8, marginLeft: 8 }}>Buy Me A Coffee</Text>
          </View>

          <Divder margin={5} />

          <View style={style.rowContainer}>
            <RadioButton
              color={style.colorPrimary.color}
              value="places"
              status={radioButtonValue === 'stripe' ? 'checked' : 'unchecked'}
              onPress={() => handleRadioButton('stripe')}
            />
            <Text style={{ marginTop: 8, marginLeft: 8 }}>Stripe</Text>
          </View>

          <Divder margin={8} />
          {renderResultDiv()}
        </Card>
      </View>
    );

    if (!_.isEmpty(route) && !_.isEmpty(route.params) && !_.isEmpty(route.params.id)) {
      result = (
        <View>
          <RestaurantDetails navigation={navigation} id={route.params.id} />
        </View>
      );
    }

    return result;
  }

  const handleDismissSnackBar = () => {
    setSnackBarStatus(false);
  }

  return (
    <ScrollView style={style.scrollViewContainer}>
      {renderDiv()}
      <Snackbar
        visible={snackBarStatus}
        onDismiss={handleDismissSnackBar}
        action={{
          label: 'Close',
          onPress: () => {
            setSnackBarStatus(false);
          },
        }}
      >
        {snackBarMessage}
      </Snackbar>
    </ScrollView>
  );
}

export default ContactUs;
